### ISSUE TITLE
Controlling Robots with Behavior Trees 

### ISSUE THREAD

**Issue by gabrielsr** [lesunb/HMRSsim#8]
As a Simulation creator
I want to assign a behavior tree to a robot
In order to control the robot

A popular way of specifying agents' behavior in games and robotics is 'Behavior Trees'. Would be nice to have a running example of integrating BTs control in our ECS architecture and a wiki page description. 

Intro to BTs:
https://outforafight.wordpress.com/2014/07/15/behaviour-behavior-trees-for-ai-dudes-part-1/
https://arxiv.org/pdf/2005.05842.pdf
https://www.researchgate.net/publication/340870872_An_event-driven_behavior_trees_extension_to_facilitate_non-player_multi-agent_coordination_in_video_games

**Comment by gabrielsr** [comment]
Articles about how to inetgrate ECS and BTs:

https://medium.com/@icex33/entity-component-system-and-game-ai-techniques-f439eb69b5d2
Questions in this article:
- Should the tree be triggered on every tick?
- Should we support the running node state and while triggering the tree, it should jump directly to the last running node?
- Should the nodes operate on a group of entities, or on a single entity?
- Should we make trees run in parallel for every entity by using tasks?
- How should the tree be represented (in the editor)?
- Should the tree nodes have their own parameters / state, or should everything be defined with components and entities?
-

**Comment by gabrielsr** [comment]
Libs: 
https://pypi.org/project/async-btree/
https://github.com/ToyotaResearchInstitute/task_behavior_engine
